Gems/ONNX/Code/CMakeLists.txt

```diff
@@ -0,0 +1,249 @@
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+            "${Onnx_LIBS_DIR}/onnxruntime.lib"
+            Gem::ImguiAtom.Static
+            Gem::ImGui.Static
+            Gem::ImGui.ImGuiLYUtils
+)
+
+# Here add ONNX target, it depends on the Private Object library and Public API interface
+ly_add_target(
+    NAME ONNX ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAMESPACE Gem
+    FILES_CMAKE
+        onnx_shared_files.cmake
+        ${pal_dir}/onnx_shared_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            Gem::ONNX.API
+            "${Onnx_LIBS_DIR}/onnxruntime.lib"
+        PRIVATE
+            Gem::ONNX.Private.Object
+        INTERFACE
+            Gem::ImguiAtom.Static
+            Gem::ImGui.Static
+            Gem::ImGui.ImGuiLYUtils
+)
+
+# By default, we will specify that the above target ONNX would be used by
+# Client and Server type targets when this gem is enabled.  If you don't want it
+# active in Clients or Servers by default, delete one of both of the following lines:
+ly_create_alias(NAME ONNX.Clients NAMESPACE Gem TARGETS Gem::ONNX)
+ly_create_alias(NAME ONNX.Servers NAMESPACE Gem TARGETS Gem::ONNX)
+
+# For the Client and Server variants of ONNX Gem, an alias to the ONNX.API target will be made
+ly_create_alias(NAME ONNX.Clients.API NAMESPACE Gem TARGETS Gem::ONNX.API)
+ly_create_alias(NAME ONNX.Servers.API NAMESPACE Gem TARGETS Gem::ONNX.API)
+
+# If we are on a host platform, we want to add the host tools targets like the ONNX.Editor MODULE target
+if(PAL_TRAIT_BUILD_HOST_TOOLS)
+    # The ONNX.Editor.API target can be used by other gems that want to interact with the ONNX.Editor module
+    ly_add_target(
+        NAME ONNX.Editor.API INTERFACE
+        NAMESPACE Gem
+        FILES_CMAKE
+            onnx_editor_api_files.cmake
+            ${pal_dir}/onnx_editor_api_files.cmake
+        INCLUDE_DIRECTORIES
+            INTERFACE
+                Include
+        BUILD_DEPENDENCIES
+            INTERFACE
+                AZ::AzToolsFramework
+                Gem::ImguiAtom.Static
+                Gem::ImGui.Static
+                Gem::ImGui.ImGuiLYUtils
+    )
+
+    # The ONNX.Editor.Private.Object target is an internal target
+    # which is only to be used by this gems CMakeLists.txt and any subdirectories
+    # Other gems should not use this target
+    ly_add_target(
+        NAME ONNX.Editor.Private.Object STATIC
+        NAMESPACE Gem
+        FILES_CMAKE
+            onnx_editor_private_files.cmake
+        TARGET_PROPERTIES
+            O3DE_PRIVATE_TARGET TRUE
+        INCLUDE_DIRECTORIES
+            PRIVATE
+                Include
+                Source
+        BUILD_DEPENDENCIES
+            PUBLIC
+                AZ::AzToolsFramework
+                $<TARGET_OBJECTS:Gem::ONNX.Private.Object>
+                "${Onnx_LIBS_DIR}/onnxruntime.lib"
```

Comment:
repeat: This has to a CMake target and not a path to a library.

Reply:
Changed